wurst/systems/experience/Experience.wurst

```diff
@@ -32,11 +36,30 @@ init
     for i = 0 to PLAYER_NEUTRAL_PASSIVE - 1
         players[i].setHandicapXP(0)
 
+    doPeriodically(ANIMATION_PERIOD) (cb) ->
+        for troll in udg_trolls
+            let owner = troll.getOwner()
+            if troll.getMana() < 1 or owner.getGold() < 1 and troll.isAlive()

```

Comment:
What are your thoughts on adding ITT-specific extentions for `troll.getEnergy` and `troll.getHeat`. This will allow the code to be more modular if we ever change these systems.

Reply:
I don't mind `getHeat` but `getEnergy` seems overkill

Reply:
Could be. It'd be a pretty drastic change to necessitate that, so I'm fine without it.